### ISSUE TITLE
Document how to test internationalised components for v2

### ISSUE THREAD

**Issue by alejandroiglesias** [yahoo/react-intl#316]
Hi, I would like to make this suggestion, to request you to please document how to test a component that uses React Intl, since that was a point of friction that I couldn't actually resolve yet after many days (see #308 and #315).
Many thanks in advance.

**Comment by ericf** [comment]
I added some docs about how to write shallow rendering tests for components that use React Intl: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl

I'll work on other types of testing when I have time…

**Comment by alejandroiglesias** [comment]
Thanks a lot!

**Comment by emj365** [comment]
Could you write some demos for JEST to test `setState`?

**Comment by ericf** [comment]
@emj365 this is not a priority. Instead of asking _me_ to do this, you should learn how to use the tools and frameworks you use for your app.

**Comment by emj365** [comment]
@ericf OK I got it "is not a priority". Sorry for asking you. But I suggest it here. Thanks anyway.

**Comment by ericf** [comment]
@emj365 you're asking for support for the same issue in at least three different issues. I've never used Jest, I'd have to learn it to create a demo, you're better off learning it yourself than waiting on me to create a demo for your particular, generic, non-react-intl-specific, use case.

**Comment by emj365** [comment]
@ericf sorry again. No JEST demo is OK. I suggest more test demos other than Shallow render in future. Thank you.

**Comment by ericf** [comment]
That's already mentioned here: https://github.com/yahoo/react-intl/issues/316#issuecomment-184510314

**Comment by ericf** [comment]
Closing this. People should contribute Jest and Enzyme docs if they use them, I don't so it's a low priority for _me_ to learn, then document these.

**Comment by joncursi** [comment]
Can anyone provide some insight on how to create an enzyme helper function that simply wraps a component inside `<IntlProvider />`?

Helper:

```jsx
import { IntlProvider } from 'react-intl';
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

const enTranslationData = require('../../i18n/en.json');

function renderWithIntl(component, locale) {
  return (
    ReactTestUtils.renderIntoDocument(
      <IntlProvider
        locale={locale}
        messages={enTranslationData}
      >
        {React.cloneElement(component)}
      </IntlProvider>
    )
  );
}

export default renderWithIntl;
```

Test:

```jsx
  it('has the correct content', () => {
    const wrapper = renderWithIntl(<NotFoundPage />, 'en-US');
    expect(wrapper).to.equal('Page not found');
  });
```

Assertion Error:

```
     AssertionError: expected { Object (props, context, ...) } to equal 'Page not found'
```

When I try to get the text from it:

```
    expect(wrapper.text()).to.equal('Page not found');
```

I get another assertion error:

```
     TypeError: wrapper.text is not a function
```
